File: android/app/src/main/java/model/Barbell.java
package model;

public class Barbell extends Exercise {

	private double weight;
	
	public Barbell(String name, String musclegroup) {
		super (name, musclegroup);
	}
	
	/**
	 * returns the current weight that the user is lifting
	 * 
	 * @return the weight
	 */
	public double getWeight() {
		return weight;
	}

	/**
	 * sets the weight to 10 lbs more than the previous weight
	 * 
	 */
	public void updateWeighthard() {
		weight+=10;
	}
	
	/**
	 * sets the weight to 5 lbs more than the previous weight
	 * 
	 */
	public void updateWeightsoft() {
		weight+=10;
	}
	
	/**
	 * sets the weight to a given weight
	 * @param weight in lbs
	 */
	public void setWeight(int weight) {
		this.weight= weight;
	}
	
}

File: lib/algo/Barbell.dart
import 'Exercise.dart';

class Barbell extends Exercise
{
  int weight = 0;

  Barbell(String name, String musclegroup) : super(name, musclegroup)
  {

  }

  int getWeight()
  {
    return weight;
  }

  void updateWeighthard()
  {
    weight += 10;
  }

  void updateWeightsoft()
  {
    weight += 10;
  }

  void setWeight(int weight)
  {
    this.weight = weight;
  }
}